File: 03-ordenacao/CPP/principal.cpp
#include <iostream>
#include <time.h>

#include "metodosOrdenacao.h"
#include "utilidades.h"

using namespace std;

int main() {
    clock_t tempoInicio, tempoFim;
    long long int tamanho;

    system("clear");//no windows é system("cls");

    cout << "Tamanho do vetor: ";
    cin >> tamanho;
    int *vetorBolha;
    int *vetorSelecao;
    int *vetorPente;
    vetorBolha = (int*)malloc(sizeof(int) * tamanho);
    vetorSelecao = (int*)malloc(sizeof(int) * tamanho);
    vetorPente = (int*)malloc(sizeof(int) * tamanho);
    popular(vetorBolha, tamanho);
    copiar(vetorBolha, tamanho, vetorSelecao);
    copiar(vetorBolha, tamanho, vetorPente);
    //exibir(vetorSelecao, tamanho);

    //pente
    tempoInicio = clock();
    pente(vetorPente,tamanho);
    tempoFim = clock();
    cout << "Tempo para pente: "<< (float)(tempoFim - tempoInicio)/CLOCKS_PER_SEC << endl;
    //exibir(vetorPente, tamanho);

    // //bolha
    // tempoInicio = clock();
    // bolha(vetorBolha,tamanho);
    // tempoFim = clock();
    // cout << "Tempo para bolha: "<< (float)(tempoFim - tempoInicio)/CLOCKS_PER_SEC << endl;
    // // exibir(vetorBolha, tamanho);

    // //selecao
    // tempoInicio = clock();
    // selecao(vetorSelecao,tamanho);
    // tempoFim = clock();
    // cout << "Tempo para selecao: "<< (float)(tempoFim - tempoInicio)/CLOCKS_PER_SEC << endl;
    // //exibir(vetorSelecao, tamanho);

  

    return 1;
}

File: 03-ordenacao/Python/principal.py
import time
import os

from util import Util
from ordenacao import Ordenacao

lista_bolha = []
lista_normal = []
lista_selecao = []
lista_insercao = []
lista_agitacao = []
lista_pente = []

tamanho = 30000

Util.popular_lista_aleatoria(lista_bolha, tamanho, 1000, 20000)
lista_normal.extend(lista_bolha)
lista_selecao.extend(lista_bolha)
lista_insercao.extend(lista_bolha)
lista_agitacao.extend(lista_bolha)
lista_pente.extend(lista_bolha)

os.system('clear')

#sort
tempoInicio = time.perf_counter()
lista_normal.sort()
tempoFim = time.perf_counter()
tempo_sort = tempoFim - tempoInicio
print("Tempo da rotina ordenar por sort nativo: ",  tempo_sort, "s")       

# #pente
tempo_inicio = time.perf_counter()
qtd_comparacoes, qtd_trocas = Ordenacao.pente(lista_pente)
tempo_fim = time.perf_counter()
print("Tempo da rotina ordenar por pente: ", (tempo_fim - tempo_inicio) , "s")      
print('Comparacoes:', qtd_comparacoes)
print('Trocas:', qtd_trocas)

#bolha
tempoInicio = time.perf_counter()
qtd_comparacoes, qtd_trocas = Ordenacao.bolha(lista_bolha)
tempoFim = time.perf_counter()
tempo_bolha = tempoFim - tempoInicio
print("Tempo da rotina ordenar por bolha: ", tempo_bolha , "s")      
print('Comparacoes:', qtd_comparacoes)
print('Trocas:', qtd_trocas)  

#selecao
tempoInicio = time.perf_counter()
qtd_comparacoes, qtd_trocas = Ordenacao.selecao(lista_selecao)
tempoFim = time.perf_counter()
tempo_selecao = tempoFim - tempoInicio
print("Tempo da rotina ordenar por selecao: ", tempo_selecao , "s")      
print('Comparacoes:', qtd_comparacoes)
print('Trocas:', qtd_trocas)  

#insercao
tempoInicio = time.perf_counter()
qtd_comparacoes, qtd_trocas = Ordenacao.insercao(lista_insercao)
tempoFim = time.perf_counter()
tempo_insercao = tempoFim - tempoInicio
print("Tempo da rotina ordenar por insercao: ", tempo_insercao , "s")      
print('Comparacoes:', qtd_comparacoes)
print('Trocas:', qtd_trocas)  

#agitacao
tempo_inicio = time.perf_counter()
qtd_comparacoes, qtd_trocas = Ordenacao.agitacao(lista_agitacao)
tempo_fim = time.perf_counter()
print("Tempo da rotina ordenar por agitacao: ", (tempo_fim - tempo_inicio) , "s")      
print('Comparacoes:', qtd_comparacoes)
print('Trocas:', qtd_trocas)    

  

